templates/s3-redirector.yaml

```diff
@@ -0,0 +1,103 @@
+# This template sets up an S3 static website with a routing rule to redirect (307)
+# to another website, optionally replacing the key. Also sets up a Cloudfront
+# distribution to handle https.
+AWSTemplateFormatVersion: 2010-09-09
+Description: Provision a S3 static website redirector with Cloudfront
+Parameters:
+  SourceHostName:
+    Description: The source endpoint (to redirect from)
+    Type: String
+    AllowedPattern: (?!-)[a-zA-Z0-9-.]{1,63}(?<!-)
+    ConstraintDescription: must be a valid DNS zone name.
+  SourceAcmCertificateArn:
+    Description: The Amazon Resource Name (ARN) of an AWS Certificate Manager (ACM) certificate.
+    Type: String
+    AllowedPattern: "arn:aws:acm:.*"
+    ConstraintDescription: must be a valid certificate ARN.
+  TargetHostName:
+    Type: String
+    Description: The target endpoint
+    AllowedPattern: (?!-)[a-zA-Z0-9-.]{1,63}(?<!-)
+    ConstraintDescription: must be a valid DNS zone name.
```

Comment:
so this is the VPN SSP endpoint.  does this mean we need to also create a hosted zone for it?

Reply:
My understanding is that it will reside in the sageit.org domain so all we need to do is add a CNAME that will point to the URL of the Cloudfront distribution (btw to answer your first comment, we only need Cloudfront to handle https).
My thinking was to keep this one as simple as possible (there are cases where the CNAME already exists and just needs to be updated, and not always created with Cloudformation). For our application, the plan was to create a separate template to create the CNAME pointing to the CloudfrontEndpoint of this template. Come to think of it, maybe I should make the CNAME creation optional in this one (i.e. create if specified)?